### PR TITLE
Extending MD_Keywords for where the keyword is a gmx:Anchor

### PR DESCRIPTION
**Pull request by adamml**
The Thesaurus had already been updated for this, but I have now added a case for keywords coming from a gmx:Anchor.

To facilitate this:

- I added a keyword attribute to MD_Keywords
- The MD_Keywords.keyword attribute is a list, and is populated by instances of a new class, keyword
- keyword has two attributes, name and URL 
- keyword is created for both cases of gmd:keyword - where the content is a gco:CharacterString and the url is therefore None, and for when the content is a gmx:Anchor and the url is defined by the href

### CODE REVIEW

File: owslib/iso.py [hunk 1]
```diff
@@ -298,22 +302,39 @@ def __init__(self, md=None):
             self.role = _testCodeListValue(md.find(util.nspath_eval('gmd:role/gmd:CI_RoleCode', namespaces)))
 
 
+class keyword(object):
```
Comment (huard) [inline owslib/iso.py:305]: Use CamelCase for class names.

**Comment by tomkralidis** [comment]
@adamml some comments:
- there are many parts of the commit with additions of blank lines and changes to `#noqa` markers.  Was this intended?
- this is a valuable enhancement given the functionality of anchors in ISO keywords.  We should really be planning to change the nature of `owslib.ogcapi.iso.MD_Keywords.keywords` from a list of strings to a list of objects. I would suggest changing the name `keyword` to `keywords_objects` and add a warning when initializing the `MD_Keywords` class that `keywords_objects` will become `keywords` proper in the next release

**Comment by adamml** [comment]
@huard @tomkralidis Thanks for the review and the comments. I'll try and address them and comment back here

File: owslib/iso.py (hunk 1 above)
Comment (adamml) [reply owslib/iso.py]: Camel Case added in 1d5df8a, I will look to add a test now too

Comment (adamml) [reply owslib/iso.py]: Tests added in 72c4148

**Comment by adamml** [comment]
@huard, @tomkralidis Thanks again for reviewing the PR and making suggestions. I think I've addressed them all in a number of commits to my fork today. Let me know what you think and if there are any more steps to take before this PR can be merged.

**Pull request merged**